hasjob/views/index.py

```diff
@@ -307,16 +299,20 @@ def fetch_jobposts(request_args, request_values, is_index, board, board_jobs, gk
     if loadmore:
         query_params.update({'startdate': loadmore.isoformat() + 'Z', 'ph': pinned_hashids})
     if location:
+        data_filters['location_names'].append(location['name'])
         query_params.update({'l': location['name']})
 
     if pay_graph:
         pay_graph_data = make_pay_graph(pay_graph, posts, rmin=f_min, rmax=f_max)
 
+    postids = [post.id for post in posts]
+    data_filters['tags'] = Tag.filter_by_posts(postids)
+    data_filters['domains'] = Domain.filter_by_posts(postids)
```

Comment:
These are expensive queries. They are also unnecessary -- our use case is to limit jobs to the specified tags and domains, not to retrieve the full set of tags and domains.